components/cdc/src/initializer.rs

```diff
@@ -292,15 +322,33 @@ impl<E: KvEngine> Initializer<E> {
         let mut stats = Statistics::default();
         while total_bytes <= self.max_scan_batch_bytes && total_size < self.max_scan_batch_size {
             total_size += 1;
-            match scanner.next_entry()? {
-                Some(mut entry) => {
-                    read_old_value(entry.old_value(), &mut stats)?;
-                    total_bytes += entry.size();
-                    entries.push(Some(entry));
-                }
-                None => {
-                    entries.push(None);
-                    break;
+            if let Some(scanner_iter) = scanner_iter {
+                match scanner_iter {
+                    ScannerIter::Scanner(scanner) => {
```

Comment:
Suggest use generic type or function to separate the two implementations. If-else or match will lead to difficulty of code maintainance in the future.